src/libs/TransactionUtils.ts

```diff
@@ -0,0 +1,358 @@
+            return Math.abs(amount);
+        }
+        return Math.abs(transaction?.amount ?? 0);
+    }
+
+    // Expense report case:
+    // The amounts are stored using an opposite sign and negative values can be set,
+    // we need to return an opposite sign than is saved in the transaction object
+    let amount = transaction?.modifiedAmount ?? 0;
+    if (amount) {
+        return -amount;
+    }
+
+    // To avoid -0 being shown, lets only change the sign if the value is other than 0.
+    amount = transaction?.amount ?? 0;
+    return amount ? -amount : 0;
+}
+
+/**
+ * Return the currency field from the transaction, return the modifiedCurrency if present.
+ */
+function getCurrency(transaction: Transaction): string {
+    const currency = transaction?.modifiedCurrency ?? '';
+    if (currency) {
+        return currency;
+    }
+    return transaction?.currency ?? CONST.CURRENCY.USD;
+}
+
+/**
+ * Return the merchant field from the transaction, return the modifiedMerchant if present.
+ */
+function getMerchant(transaction: Transaction): string {
+    return transaction?.modifiedMerchant ?? transaction?.merchant ?? '';
+}
+
+/**
+ * Return the category from the transaction. This "category" field has no "modified" complement.
+ */
+function getCategory(transaction: Transaction): string {
+    return transaction?.category ?? '';
+}
+
+/**
+ * Return the created field from the transaction, return the modifiedCreated if present.
+ */
+function getCreated(transaction: Transaction): string {
+    const created = transaction?.modifiedCreated ?? transaction?.created ?? '';
+    const createdDate = parseISO(created);
+    if (isValid(createdDate)) {
+        return format(createdDate, CONST.DATE.FNS_FORMAT_STRING);
+    }
+
+    return '';
+}
+
+function isDistanceRequest(transaction: Transaction): boolean {
+    const type = transaction?.comment?.type;
+    const customUnitName = transaction?.comment?.customUnit?.name;
+    return type === CONST.TRANSACTION.TYPE.CUSTOM_UNIT && customUnitName === CONST.CUSTOM_UNITS.NAME_DISTANCE;
+}
+
+function isReceiptBeingScanned(transaction: Transaction): boolean {
+    return [CONST.IOU.RECEIPT_STATE.SCANREADY, CONST.IOU.RECEIPT_STATE.SCANNING].some((value) => value === transaction.receipt.state);
+}
+
+/**
+ * Check if the transaction has a non-smartscanning receipt and is missing required fields
+ */
+function hasMissingSmartscanFields(transaction: Transaction): boolean {
+    return hasReceipt(transaction) && !isDistanceRequest(transaction) && !isReceiptBeingScanned(transaction) && areRequiredFieldsEmpty(transaction);
+}
+
+/**
+ * Check if the transaction has a defined route
+ */
+function hasRoute(transaction: Transaction): boolean {
+    return !!transaction?.routes?.route0?.geometry?.coordinates;
+}
+
+/**
+ * Get the transactions related to a report preview with receipts
+ * Get the details linked to the IOU reportAction
+ *
+ * @deprecated Use Onyx.connect() or withOnyx() instead
```

Comment:
Looks like this line is deprecated entirely 😅
```suggestion
```

Reply:
@rezkiy37 it was added to this js file with this pr https://github.com/Expensify/App/pull/27286 a week ago inside this commit https://github.com/Expensify/App/pull/27286/commits/0102215f4b4fc54c3ed74215122ce3a8d7b7c495

Reply:
Ahhhhhhh, I got their idea. They mean that the function is deprecated, I thought that `Onyx.connect()` is deprecated. Forget :)

Reply:
But it is still used in the code @rezkiy37